### ISSUE TITLE
Idea's for smaller changes: EKS Cluster Module

### ISSUE THREAD

**Issue by qvistgaard** [nuuday/terraform-aws-eks#3]
Basically i like the work that has been done so far, i just wish that i would be more modular, it would make it easier to reuse:

my initial suggestion is to create submodules for a lot of the features. i being split into the following submodules:

**The cluster**
I would love to see that the eks cluster configuration mirror it's parent module. This should mean that all variable should also be the same. That being said, i like the idea that we can just  provide `linux_workers_count` and that's what you will get. However I think it will limit it a bit.

If possible i would love to see something like this, if possible, this would also make it possible to allow others to do more custom stuff:
```
module "eks" "test" {
   ....
   worker_groups: [ module.eks_worker.workers.output.json ]
}

module "eks_worker" "workers" {
   ....
}
```

**Comment by anarsen** [comment]
Completely agree. I'd prefer if we added a variable of the same type as https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf#L34

The simple `linux_workers_count` is taken directly from the sample and needs changing :rocket:

**Comment by qvistgaard** [comment]
Ok, i will work on a pull request then

**Comment by anarsen** [comment]
We did a call and agreed the `nuuday/terraform-aws-eks` will be "API compatible" with the community-driven `terraform-aws-modules/terraform-aws-eks` module.

@qvistgaard Can we close this issue, or perhaps reword it to fit the above?

**Comment by qvistgaard** [comment]
Let's close it
